### PR TITLE
DRILL-6900: maven-release-plugin failure on the release:perform stage

### PR DESCRIPTION
**Pull request by vdiravka**
- downgrade maven-javadoc-plugin version
- update some Drill maven plugins versions and place them to pluginManagement block
- bump up lowest maven mevrsion supported by Drill in correspondence to org.apache.maven dependencies

### CODE REVIEW

**Comment by vdiravka** [comment]
@sohami Could you please review? It is slightly related to recent changes for updating `apache-21.pom` parent POM file.

I have described the issue in the Jira ticket and solution in my [comment](https://issues.apache.org/jira/browse/DRILL-6900?focusedCommentId=16720707&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-16720707) in Jira too.
The only one change from this PR, which fixes the issue is a downgrade of `maven-javadoc-plugin` version. I have created separate ticket [DRILL-6904](https://issues.apache.org/jira/browse/DRILL-6904) to update this plugin in future.
All other changes I made in the process of analysis the issue (mainly it is updating maven plugins versions). 
Please take a look, whether it is fine. If not, I can create a separate ticket for updating maven plugins and include it to 1.16.0 then.

File: pom.xml [hunk 1]
```diff
@@ -89,11 +89,12 @@
     <excludedGroups />
     <memoryMb>4096</memoryMb>
     <directMemoryMb>4096</directMemoryMb>
-    <doclint>none</doclint>
+    <additionalparam>-Xdoclint:none</additionalparam>
```
Comment (sohami) [inline pom.xml]: Please put this parameter [here](https://github.com/apache/drill/pull/1574/files#diff-600376dffeb79835ede4a0b285078036R907)

Comment (vdiravka) [reply pom.xml]: It makes sense. Done

File: docs/dev/Environment.md [hunk 2]
```diff
@@ -13,7 +13,7 @@ Currently, the Apache Drill build process is known to work on Linux, Windows and
     Java HotSpot(TM) 64-Bit Server VM (build 25.161-b12, mixed mode)
 
     # mvn --version
-    Apache Maven 3.3.1 (cab6659f9874fa96462afef40fcf6bc033d58c1c; 2015-03-13T13:10:27-07:00)
+    Apache Maven 3.6.0 (97c98ec64a1fdfee7767ce5ffb20918da4f719f3; 2018-10-24T21:41:47+03:00)
```
[the remark this answers is not in the document]
Comment (vdiravka) [reply docs/dev/Environment.md]: right, I leave `Apache Maven 3.3.3`. here

**Comment by vdiravka** [comment]
@sohami Could you please take a look?

**Pull request closed without merging**